pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -176,52 +191,53 @@ func (f *FederatedTypeConfig) GetObjectMeta() metav1.ObjectMeta {
 	return f.ObjectMeta
 }
 
-func (f *FederatedTypeConfig) GetTarget() metav1.APIResource {
-	return apiResourceToMeta(f.Spec.Target, f.Spec.Namespaced)
+func (f *FederatedTypeConfig) GetTargetType() metav1.APIResource {
+	return apiResourceToMeta(f.Spec.TargetType, f.GetNamespaced())
 }
 
 func (f *FederatedTypeConfig) GetNamespaced() bool {
-	return f.Spec.Namespaced
+	return f.Spec.TargetType.Namespaced()
 }
 
 func (f *FederatedTypeConfig) GetPropagationEnabled() bool {
-	return f.Spec.PropagationEnabled
+	return f.Spec.Propagation == PropagationEnabled
 }
 
 func (f *FederatedTypeConfig) GetFederatedType() metav1.APIResource {
 	return apiResourceToMeta(f.Spec.FederatedType, f.GetFederatedNamespaced())
 }
 
-func (f *FederatedTypeConfig) GetStatus() *metav1.APIResource {
-	if f.Spec.Status == nil {
+func (f *FederatedTypeConfig) GetStatusType() *metav1.APIResource {
+	if f.Spec.StatusType == nil {
 		return nil
 	}
-	metaAPIResource := apiResourceToMeta(*f.Spec.Status, f.Spec.Namespaced)
+	metaAPIResource := apiResourceToMeta(*f.Spec.StatusType, f.Spec.StatusType.Namespaced())
 	return &metaAPIResource
 }
 
-func (f *FederatedTypeConfig) GetEnableStatus() bool {
-	return f.Spec.EnableStatus
+func (f *FederatedTypeConfig) GetStatusEnabled() bool {
+	return f.Spec.StatusCollection != nil && *f.Spec.StatusCollection == StatusCollectionEnabled
 }
 
-// TODO(marun) Remove in favor of using 'true' for namespaces and the
-// value from target otherwise.
 func (f *FederatedTypeConfig) GetFederatedNamespaced() bool {
```

Comment:
This function should be removed in favor of checking the namespaced property of calling `Namespaced` on the federated apiresource.